dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPI.java

```diff
@@ -92,9 +76,19 @@ Metadata getMetadata(Contentlet contentlet, String fieldVariableName)
      * @param fieldVariableName  {@link String}
      * @return Map
      */
-     Metadata getMetadataForceGenerate(Contentlet contentlet, String fieldVariableName)
+     Metadata getOrGenerateMetadata(Contentlet contentlet, String fieldVariableName)
             throws DotDataException;
 
+    /**
+     * This number helps us to determine if we need to regenerate lazily the md if we want to include new attributes on the md
+     * @return ver number follows the same pattern as our StartupTasks, eg, the date formatted like YYYYMMDD
+     */
+    default int getBinaryMetadataVersion(){
+        //We should always return the hard coded val but I'm leaving it open so it can be override via props
```

Comment:
if this is read a lot I would use a Lazy approach

Reply:
That's  actually a good idea